codegen/src/Azure.Iot.Operations.ProtocolCompilerLib/T4/communication/rust/Property/t4/RustPropertyConsumer.tt

```diff
@@ -0,0 +1,439 @@
+pub struct <#=this.propertyName.GetTypeName(TargetLanguage.Rust, "write", "request", "builder")#> {
+    inner_builder: rpc_command::invoker::RequestBuilder<<#=this.writeReqSchema.GetTypeName(TargetLanguage.Rust)#>>,
+<# if (this.doesPropertyTargetMaintainer) { #>
+    set_maintainer_id: bool,
+<# } #>
+    topic_tokens: HashMap<String, String>,
+}
+
+impl <#=this.propertyName.GetTypeName(TargetLanguage.Rust, "write", "request", "builder")#> {
+    /// Custom user data to set on the write request
+    pub fn custom_user_data(&mut self, custom_user_data: Vec<(String, String)>) -> &mut Self {
+        self.inner_builder.custom_user_data(custom_user_data);
+        self
+    }
+
+    /// Topic token keys/values to be replaced into the publish topic of the request message.
+    /// A prefix of "ex:" will be prepended to each key before scanning the topic pattern.
+    /// Thus, only tokens of the form `{ex:SOMEKEY}` will be replaced.
+    pub fn topic_tokens(&mut self, topic_tokens: HashMap<String, String>) -> &mut Self {
+        for (k, v) in topic_tokens {
+            self.topic_tokens.insert(format!("ex:{k}"), v);
+        }
+        self
+    }
+
+    /// Timeout for the request
+    pub fn timeout(&mut self, timeout: Duration) -> &mut Self {
+        self.inner_builder.timeout(timeout);
+        self
+    }
+
+<# if (this.doesPropertyTargetMaintainer) { #>
+    /// Target maintainer ID
+    pub fn maintainer_id(&mut self, maintainer_id: &str) -> &mut Self {
+        self.topic_tokens
+            .insert("maintainerId".to_string(), maintainer_id.to_string());
+        self.set_maintainer_id = true;
+        self
+    }
+
+<# } #>
+    /// Payload of the write request
+    ///
+    /// # Errors
+    /// If the payload cannot be serialized
+    pub fn payload(
+        &mut self,
+        payload: <#=this.writeReqSchema.GetTypeName(TargetLanguage.Rust)#>,
+    ) -> Result<&mut Self, AIOProtocolError> {
+        self.inner_builder.payload(payload)?;
+        Ok(self)
+    }
+
+    /// Builds a new `<#=this.propertyName.GetTypeName(TargetLanguage.Rust, "write", "request")#>`
+    ///
+    /// # Errors
+    /// If a required field has not been initialized
+    #[allow(clippy::missing_panics_doc)]    // The panic is not possible
+    pub fn build(&mut self) -> Result<<#=this.propertyName.GetTypeName(TargetLanguage.Rust, "write", "request")#>, <#=this.propertyName.GetTypeName(TargetLanguage.Rust, "write", "request", "builder", "error")#>> {
+<# if (this.doesPropertyTargetMaintainer) { #>
+        if !self.set_maintainer_id {
+            return Err(<#=this.propertyName.GetTypeName(TargetLanguage.Rust, "write", "request", "builder", "error")#>::UninitializedField(
+                "maintainer_id",
+            ));
+        }
+
+<# } #>
+        self.inner_builder.topic_tokens(self.topic_tokens.clone());
+
+        self.inner_builder.build()
+    }
+}
+<# } #>
+
+/// Read requester for `<#=this.propertyName.AsGiven#>`
```

Comment:
this (and similar on the maintainer side) is generating as 
> /// Read requester for ``

for the property_together sample

Reply:
Fixed.